### PR TITLE
contrib/google.golang.org/grpc: security rule passlist support

### PR DESCRIPTION
**Pull request by Julio-Guerra**
### What does this PR do?

Implement the `grpc.server.method` security rule address.
Tested with a custom ruleset passlist.json.

### Motivation

AppSec Passlist Support for gRPC, where a gRPC security rule can be passlisted according to the gRPC method name.

### Reviewer's Checklist

- [x] Changed code has unit tests for its functionality at or near 100% coverage.
- [x] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [x] There is a benchmark for any new code, or changes to existing code.
- [x] If this interacts with the agent in a new way, a system test has been added.
- [x] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [x] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

For Datadog employees:

- [x] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

APPSEC-52066

### CODE REVIEW

File: internal/appsec/listener/grpcsec/grpc.go [hunk 1]
```diff
@@ -100,35 +103,36 @@ func (l *wafEventListener) onEvent(op *types.HandlerOperation, handlerArgs types
...
-	// OnUserIDOperationStart happens when appsec.SetUser() is called. We run the WAF and apply actions to
-	// see if the associated user should be blocked. Since we don't control the execution flow in this case
-	// (SetUser is SDK), we delegate the responsibility of interrupting the handler to the user.
-	dyngo.On(op, func(userIDOp *sharedsec.UserIDOperation, args sharedsec.UserIDOperationArgs) {
-		values := make(map[string]any, 1)
-		for addr := range l.addresses {
-			if addr == UserIDAddr {
-				values[UserIDAddr] = args.UserID
+	// Listen to the UserID address if the WAF rules are using it
+	if l.isSecAddressListened(UserIDAddr) {
```
Comment (RomainMuller) [inline internal/appsec/listener/grpcsec/grpc.go:107]: Not gonna say anything, but this change seems unrelated to the passlist support 🫣

File: internal/appsec/listener/grpcsec/grpc.go [hunk 2]
```diff
@@ -100,35 +103,36 @@ func (l *wafEventListener) onEvent(op *types.HandlerOperation, handlerArgs types
...
-	// The same address is used for gRPC and http when it comes to client ip
-	values := make(map[string]any, 1)
-	for addr := range l.addresses {
-		if addr == HTTPClientIPAddr && handlerArgs.ClientIP.IsValid() {
-			values[HTTPClientIPAddr] = handlerArgs.ClientIP.String()
-		}
+	values := make(map[string]any, 2) // 2 because the method and client ip addresses are commonly present in the rules
+	if l.isSecAddressListened(GRPCServerMethodAddr) {
+		values[GRPCServerMethodAddr] = handlerArgs.Method
+		// Note that this address is passed asap for the passlist, which are created per grpc method
```
Comment (RomainMuller) [inline internal/appsec/listener/grpcsec/grpc.go:132]: Completely optional, but it feels odd to me that the comment applies to what's above it... usually it's what's below it.


```suggestion
		// Note that this address is passed asap for the passlist, which are created per grpc method
		values[GRPCServerMethodAddr] = handlerArgs.Method
```

**Review by RomainMuller — APPROVED** [review]

File: internal/appsec/listener/grpcsec/grpc.go (hunk 1 above)
Comment (Julio-Guerra) [reply internal/appsec/listener/grpcsec/grpc.go:107]: Discussed at the standup: let's start relying on tools such as git machete to make it simpler to have separate refactoring PR stacks

**Review by dianashevchenko — APPROVED** [review]
